tutorials/02_intermediate/gridspec.py

```diff
@@ -1,126 +1,196 @@
-# them to create a subplot.
-#
-# For example:
-
 fig = plt.figure()
-ax = plt.subplot2grid((2, 2), (0, 0))
-
-# is equal to:
-
-fig = plt.figure()
-gs = gridspec.GridSpec(2, 2)
-ax = plt.subplot(gs[0, 0])
-
-# A GridSpec instance provides array-like (2d or 1d) indexing that
-# returns the SubplotSpec instance. For a SubplotSpec that spans multiple
-# cells, use slice. ::
-
-ax2 = plt.subplot(gs[1, :-1])
-ax3 = plt.subplot(gs[1:, -1])
-
-###############################################################################
-# The above example becomes ::
-
-fig = plt.figure()
-gs = gridspec.GridSpec(3, 3)
-ax1 = plt.subplot(gs[0, :])
-ax2 = plt.subplot(gs[1, :-1])
-ax3 = plt.subplot(gs[1:, -1])
-ax4 = plt.subplot(gs[-1, 0])
-ax5 = plt.subplot(gs[-1, -2])
+ax1 = plt.subplot2grid((3, 3), (0, 0), colspan=3, fig=fig)
+ax2 = plt.subplot2grid((3, 3), (1, 0), colspan=2, fig=fig)
+ax3 = plt.subplot2grid((3, 3), (1, 2), rowspan=2, fig=fig)
+ax4 = plt.subplot2grid((3, 3), (2, 0), fig=fig)
+ax5 = plt.subplot2grid((3, 3), (2, 1), fig=fig)
+fig.tight_layout()
 
 ###############################################################################
-# Adjust GridSpec layout
-# ======================
+# Fine Adjustments to a Gridspec Layout
+# =====================================
 #
 # When a GridSpec is explicitly used, you can adjust the layout
-# parameters of subplots that are created from the GridSpec. ::
+# parameters of subplots that are created from the GridSpec.
 
 fig = plt.figure()
-gs1 = gridspec.GridSpec(3, 3)
-gs1.update(left=0.05, right=0.48, wspace=0.05)
+gs1 = gridspec.GridSpec(nrows=3, ncols=3, left=0.05, right=0.48, wspace=0.05)
 
 ###############################################################################
 # This is similar to :func:`~matplotlib.pyplot.subplots_adjust`, but it only
 # affects the subplots that are created from the given GridSpec.
 #
-# For example, see this code and the resulting figure:
+# For example, compare the left and right sides of this figure:
 
 fig = plt.figure()
-gs1 = gridspec.GridSpec(3, 3)
-gs1.update(left=0.05, right=0.48, wspace=0.05)
-ax1 = plt.subplot(gs1[:-1, :])
-ax2 = plt.subplot(gs1[-1, :-1])
-ax3 = plt.subplot(gs1[-1, -1])
+gs1 = gridspec.GridSpec(nrows=3, ncols=3, left=0.05, right=0.48,
+                        wspace=0.05, fig=fig)
+ax1 = fig.add_subplot(gs1[:-1, :])
+ax2 = fig.add_subplot(gs1[-1, :-1])
+ax3 = fig.add_subplot(gs1[-1, -1])
 
-fig = plt.figure()
-gs2 = gridspec.GridSpec(3, 3)
-gs2.update(left=0.55, right=0.98, hspace=0.05)
-ax4 = plt.subplot(gs2[:, :-1])
-ax5 = plt.subplot(gs2[:-1, -1])
-ax6 = plt.subplot(gs2[-1, -1])
+
+gs2 = gridspec.GridSpec(nrows=3, ncols=3, left=0.55, right=0.98,
+                        hspace=0.05, fig=fig)
+ax4 = fig.add_subplot(gs2[:, :-1])
+ax5 = fig.add_subplot(gs2[:-1, -1])
+ax6 = fig.add_subplot(gs2[-1, -1])
 
 ###############################################################################
 # GridSpec using SubplotSpec
```

Comment:
here it's also be good to see some conceptual-level sentence or two about what subplotspec is, why it exists / how it's different from gridspec, etc